### ISSUE TITLE
It is much harder than it needs to be to handle errors

### ISSUE THREAD

**Issue by ghost** [martynsmith/node-irc#265]
Having one generic event for all errors with both bot-issued commands and other error events makes that error handling function a mess of if/else blocks. It'd be really cool if node-irc supported the standard method of calling the callback function with an error object if something goes wrong with the command the bot just issued.

**Comment by Jnull** [comment]
I agree, a single try catch is harder to troubleshoot than actually having error catching throughout every piece of functionality, any ideas on what structure, or how to implement this better?
